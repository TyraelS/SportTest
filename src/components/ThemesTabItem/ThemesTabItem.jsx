import React from 'react';

import ThemesTabItemStyle from './ThemesTabItemStyle';

const displayName = 'ThemesTabItem';

const tabsItemClick = (id, themesTabItemClick) => () => {
  themesTabItemClick(id);
};

const ThemesTabItem = ({ id, handler }) => {
  return (
    <ThemesTabItemStyle id={id} onClick={tabsItemClick(id, handler)}>
      {id.toUpperCase()}
    </ThemesTabItemStyle>
  );
};

ThemesTabItem.displayName = displayName;

export default ThemesTabItem;
