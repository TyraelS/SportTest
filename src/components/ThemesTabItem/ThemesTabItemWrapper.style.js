import styled from 'styled-components';

const ThemesTabItemStyle = styled.div`
  cursor: pointer;
  text-align: center;
  width: 100%;
  height: 50px;
`;

export default ThemesTabItemStyle;
