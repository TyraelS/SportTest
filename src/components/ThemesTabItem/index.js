import ThemesTabItem from './ThemesTabItem';

export default ThemesTabItem;
