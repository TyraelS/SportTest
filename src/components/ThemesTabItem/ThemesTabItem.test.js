import React from 'react';
import { shallow } from 'enzyme';

import ThemesTabItem from './ThemesTabItem';

describe('Given the ThemesTabItem component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<ThemesTabItem id={'1'} handler={jest.fn()} />);
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
