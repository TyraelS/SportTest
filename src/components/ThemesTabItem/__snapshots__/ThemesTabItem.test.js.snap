// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ThemesTabItem component when the component is rendered should match the snapshot 1`] = `
<ForwardRef(styled.div)
  id="1"
  onClick={[Function]}
>
  1
</ForwardRef(styled.div)>
`;
